activemodel/lib/active_model/load_tree.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  module LoadTree
+    def _create_load_tree_node(creator: self, parent: nil, siblings: [], child_name: nil, child_type: nil)
+      @_load_tree_node = ActiveModel::LoadTree::Node.new(
+        creator: creator,
+        parent: parent,
+        siblings: siblings,
+        child_name: child_name,
+        child_type: child_type).set_records
+    end
+
+    def _load_tree_node
+      @_load_tree_node || ActiveModel::LoadTree::Node.new(siblings: [])
+    end
+
+    class Node
+      # SiblingSizeLimit is the maximum number of siblings we will track. This
+      # Is limited because preloading can blow up if we have a lot of siblings.
+      SiblingSizeLimit = 10000
```

Comment:
Maybe this gets setup in ActiveRecord so others can set their own limit? There was a test in AR that loaded 65,000 items from the database and it was taking way too long to setup the siblings. Seems like a limit is smart.